### PR TITLE
f-checkout@v0.19.0 - webdriver tests for Selector and UserNote components

### PR DESCRIPTION
**Pull request by LTurns**
### Added
- Webdriver IO tests for dropdown in `Selector` component.
- Webdriver tests for note-field character limit in `UserNote`

### Changed
- `f-form-field` version.
- `data-test-id` references in tests.
- Loglevel to `silent` in wdio.conf

### CODE REVIEW

File: packages/f-checkout/CHANGELOG.md
Comment (xander-marjoram) [inline packages/f-checkout/CHANGELOG.md]: Webdriver 😄

Comment (LTurns) [reply packages/f-checkout/CHANGELOG.md]: hahahaa thankss

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (xander-marjoram) [inline packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Is it worth making this selector more specific, in case there are ever more dropdowns added?

File: packages/f-checkout/package.json
Comment (xander-marjoram) [inline packages/f-checkout/package.json]: Do we have a precedent for minor version bumps when there are no functional code changes? (Wondering if this should be a patch, but don't let this block the PR)

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (xander-marjoram) [inline packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: The name of this function doesn't seem quite right, maybe something like `getDropdownOptionText`?

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (xander-marjoram) [inline packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: Is the whitespace padding around the string correct?

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (LTurns) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Yeah I think that makes sense, it's very general the moment - will change it to "time-dropdown" to be more specific 👍

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (xander-marjoram) [inline packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: What does clicking on each element test? That it is clickable? If so, that might be worth pulling out into a test of its own

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (xander-marjoram) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Just to clarify I mean the CSS selector (although you can update the variable name too if you like)
```suggestion
const orderTimeDropdownOptions = () => $$('[data-test-id="form-select"] select option');
```
Would this work?

Comment (LTurns) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Ah okay I see, yep happy to change this - much clearer

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (SachaZvetelman) [inline packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Nit: indentation seems off.

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (SachaZvetelman) [inline packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: 1 second seems to be a very long pause. Is this something we must have?

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (LTurns) [reply packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: So I've just checked and the spaces are there for every option in the dropdown. I wonder if it's intentional or not - Will double check with Faye 👍

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (SachaZvetelman) [inline packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: Any particular reason to have this `Assert` here?
In this case, I think it can be at the bottom with all the other asserts.

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (LTurns) [reply packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: Nope happy to take this away 👍

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (LTurns) [reply packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: Yep exactly, I put it in here originally as an extra visual to show the user going down the list in ascending order - I could put it in a separate test though, would be equally valid and might break things down a bit

File: packages/f-checkout/package.json
Comment (LTurns) [reply packages/f-checkout/package.json]: Not that I'm aware of, but @siggerzz might know - should I change the version to 0.14.1? It's quite a small PR with no big changes so happy to change 👍

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (LTurns) [reply packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: @xander-marjoram this is being solved now by Ash and Faye. Thank you for flagging! Think I've learned a good testing lesson here 👍

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (Festivejelly) [inline packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: Remove commented out code

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (Festivejelly) [inline packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: Why are we skipping this test?

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (LTurns) [reply packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: @Festivejelly I just commented these out while I'm waiting for a small change in f-dropdown - there were some extra spaces in the dropdown text, but this should be fixed this morn so I'll uncomment and rerun the tests then 👍

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (LTurns) [reply packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: My bad, just added these to save time when running locally, will un-skip!

The 'submit checkout form' test has been skipped for awhile though because it's not quite finished - might wait until your PR is merged before I tackle that one, want to see how you've structure the consts in the page object 👍

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (siggerzz) [inline packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Can we rename this `getOrderTimeDropdownOptions`?

Comment (siggerzz) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Having said that, it might be worth rethinking this. What're your thoughts @Festivejelly?

We shouldn't be making the elements accessible from the spec.

Comment (LTurns) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Yeah I get you - seems odd to just export the html directly, not sure how else we can iterate over it in the forEach though. Could break it down so it's similar to export.inputs?

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (siggerzz) [inline packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: This should be `elements` as it's an array.

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (siggerzz) [inline packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Nit: Semicolon

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (siggerzz) [inline packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Should we rename this function to `getUserNoteLength`?

Alternatively, put this in a helper file where we can make this function more generic/reusable by passing in any selector and import it into this component object. Thoughts @Festivejelly?

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (siggerzz) [inline packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: This can be removed.

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (siggerzz) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Can we also add a JSDoc description to this function if we're going to keep it?

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (siggerzz) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Can we add JSDoc description?

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (siggerzz) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Can we add JSDoc description?

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (siggerzz) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: Can we add JSDoc description?

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (siggerzz) [inline packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: What's this refresh for?

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (LTurns) [reply packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: hahahaha yep sorry, a lot of these things I'm still editing while I'm waiting for f-dropdown to be merged, hence some weird notes and comments

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (LTurns) [reply packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: can be removed - will do that now!

Comment (LTurns) [reply packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: Don't think this test can be finished until the button leads to a new page, so we can grab the new url and show that it submits

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (LTurns) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: yep will do @siggerzz. Which functions require JSDoc descriptions? Is it any function that's more complex than the normal .isDisplayed() or .getText()?

Comment (LTurns) [reply packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: I really like the idea of having a helper file with generic functions - would be so useful for future tests 👍

File: CHANGELOG.md
Comment (ashleynolan) [inline CHANGELOG.md]: I just merged version 2.22.0, so this will need to bump to v2.22.1 I think if you rebase/merge with master again

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (siggerzz) [inline packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: I think we should only make our functions do one thing, in this case returning the user note length.

If we need to fill our the user note, it should be done in another function that's called before this one in the spec.


e.g

```
// f-checkout.spec.js

.....

CheckoutComponent.inputUserNote('foo');

var noteLength = CheckoutComponent.getUserNoteLength(); // returns: 3

expect(noteLength).toBe(3))
```

We'll also need to alter the JSDoc as a result 👍

File: packages/f-checkout/test-utils/component-objects/f-checkout.component.js
Comment (siggerzz) [inline packages/f-checkout/test-utils/component-objects/f-checkout.component.js]: I think this JSDoc should remove the reference to 'Go to payment' as the copy could always get updated in future.

I think 'Submits the checkout form' is sufficient

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js
Comment (siggerzz) [inline packages/f-checkout/test/specs/component/f-checkout.component.spec.js]: Tests on lines 15-28 can be deleted as I moved them to another spec file

**Review by siggerzz — CHANGES_REQUESTED** [review]
Just a few minor comments!

**Review by FayeCarter — DISMISSED** [review]

File: CHANGELOG.md
Comment (xander-marjoram) [inline CHANGELOG.md:6]: Do you need to bump the version in package.json too?

Comment (LTurns) [reply CHANGELOG.md:6]: done 👍

**Review by siggerzz — APPROVED** [review]
I still think the selectors could do with some work, but happy to merge for now 👍

**Review by FayeCarter — APPROVED** [review]

**Pull request merged**
